### ISSUE TITLE
elvish is not killed when terminal is closed

### ISSUE THREAD

**Issue by huiyiqun** [elves/elvish#289]
If you open and close terminal again and again, you will get many elvishes.

```
uiyiqun 21511  0.0  0.1 737836 12944 pts/1    Ssl  13:56   0:00 -elvish
huiyiqun 26272  0.0  0.2 885300 16604 pts/3    Ssl  14:03   0:00 -elvish
huiyiqun 27529  0.4  0.1 746828 15744 ?        Ssl  14:05   0:00 /usr/bin/elvish
huiyiqun 27536  0.0  0.0   4160   672 ?        S    14:05   0:00 ps aux | grep elvish
huiyiqun 27711  0.5  0.2 795172 17228 ?        Ssl  14:05   0:00 /usr/bin/elvish
huiyiqun 27717  0.0  0.0   4160   640 ?        S    14:05   0:00 ps aux | grep elvish
huiyiqun 27771  0.5  0.1 737832 12692 ?        Ssl  14:05   0:00 /usr/bin/elvish
huiyiqun 27777  0.0  0.0   4160   628 ?        S    14:05   0:00 ps aux | grep elvish
huiyiqun 27843  1.0  0.1 876844 14504 ?        Ssl  14:05   0:00 /usr/bin/elvish
huiyiqun 27849  0.0  0.0   4160   624 ?        S    14:05   0:00 ps aux | grep elvish
huiyiqun 27897  1.0  0.1 729572 11776 pts/6    Ssl  14:05   0:00 /usr/bin/elvish
huiyiqun 27904  0.0  0.0   4160   724 pts/6    S+   14:05   0:00 ps aux | grep elvish
huiyiqun 27957  0.0  0.0  12496  2156 pts/6    R+   14:05   0:00 /usr/bin/grep --color=auto elvish
```

**Comment by xiaq** [comment]
Perhaps related to handling of SIGHUP.

**Comment by huiyiqun** [comment]
This seems partially fixed by 36271874d8ddb2519739aa431564e857b5a52945.

Currently, simply starting and closing window will make elvish killed, however, if I running some external
command in elvish, it will not be killed.

For example:

```
~> sleep 100
```

then close the terminal, and open another one:

```
~> ps aux | grep sleep                                                                                                                                                    
huiyiqun  3625  0.0  0.0   8568   672 ?        S    21:08   0:00 /usr/bin/sleep 100
huiyiqun  3655  0.0  0.0  13508  2192 pts/2    S+   21:08   0:00 /usr/bin/grep --color=auto sleep
```

If I run esleep, the elvish will be killed.

**Comment by doppioandante** [comment]
I'm using Tilix as a terminal emulator and I'm experiencing the following problem:
when I close a pane running `elvish` inside of tilix, the shell doesn't seem to be killed:
```
--- SIGHUP {si_signo=SIGHUP, si_code=SI_USER, si_pid=10912, si_uid=1000} ---
futex(0x55a6470387c0, FUTEX_WAKE, 1)    = 1
rt_sigreturn({mask=[]})                 = 202
futex(0x55a64701ca70, FUTEX_WAIT, 0, NULL
```
I.E. SIGHUP is not honored as it can be seen from `strace` output.

**Comment by xiaq** [comment]
This is fixed in #494.
